### PR TITLE
Adding FixIt Metric Tool

### PR DESCRIPTION
**Pull request by hklai**
Adding a new tool to pull metric from github for FixIt.
Also added some util methods in GithubClient.

### CODE REVIEW

File: toolbox/fixit/main.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+			log.Fatalf("Error accessing user supplied token_file: %v\n", err)
+		}
+		gh = u.NewGithubClient(*org, token)
+	} else {
+		gh = u.NewGithubClientNoAuth(*org)
+	}
+}
+
+// Increment a key in the given map.
+func incrementCount(m map[string]int, key string) {
```
Comment (rkpagadala) [inline toolbox/fixit/main.go]: replace code in function with  
```
m[key]++ 
```
or get rid of function entirely

++ works and default return value is zero so it works (tested)

File: toolbox/fixit/main.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+func findIssueMetric(metric *fixItMetric) {
+	issueQueries := createIssueQuery("issue")
+	log.Printf("Issue Query: %v", issueQueries)
+
+	allIssues, err := gh.SearchIssues(issueQueries, *sort, *order)
+	if err != nil {
+		log.Printf("Failed to fetch Issues for %s: %s", *repo, err)
+		return
+	}
+	(*metric).totalIssues = len(allIssues)
```
Comment (rkpagadala) [inline toolbox/fixit/main.go]: as discussed 
```
metric.totalIssues 
```
will just work

File: toolbox/util/commonUtils.go [hunk 3]
```diff
@@ -210,3 +201,28 @@ func AssertNotEmpty(name string, value *string) {
...
+// PairList contains a list of key-value pairs.
+type PairList []Pair
+
+func (p PairList) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+func (p PairList) Len() int           { return len(p) }
+func (p PairList) Less(i, j int) bool { return p[i].Value < p[j].Value }
+
+// SortMapByValue sorts the map by the values and returns PairList.
+func SortMapByValue(m map[string]int) PairList {
+	p := make(PairList, len(m))
```
Comment (rkpagadala) [inline toolbox/util/commonUtils.go]: Change to 
```
	p := make(PairList, 0, len(m))
	for k, v := range m {
		p = append(p, Pair{k, v})
	}
```
no need for i variable

File: toolbox/release_note_collector/main.go [hunk 4]
```diff
@@ -90,47 +90,42 @@ func main() {
...
-		}
-	}
 	return nil
 }
 
-func fetchReleaseNoteFromPR(i github.Issue) (note string) {
+func fetchReleaseNoteFromPR(issue *github.Issue) (note string) {
 	reg := regexp.MustCompile("```release-note\r\n((?s).+)\r\n```")
-	m := reg.FindStringSubmatch(*i.Body)
+	m := reg.FindStringSubmatch(*(*issue).Body)
```
Comment (rkpagadala) [inline toolbox/release_note_collector/main.go]: i think you are trying not to copy the object.
the following code should work too
```
m := reg.FindStringSubmatch(*issue.Body)
```

File: toolbox/util/githubClient.go [hunk 5]
```diff
@@ -370,15 +372,20 @@ func (g GithubClient) GetTagCommitSHA(repo, tag string) (string, error) {
 
 // GetCommitCreationTime gets the time when the commit identified by sha is created
 func (g GithubClient) GetCommitCreationTime(repo, sha string) (time.Time, error) {
```
Comment (rkpagadala) [inline toolbox/util/githubClient.go]: All the methods on GithubClient can be changed to *
```
func (g *GithubClient) GetCommitCreationTime(repo, sha string) (time.Time, error) {
```

File: toolbox/fixit/main.go [hunk 6]
```diff
@@ -37,7 +37,7 @@ const (
 var (
 	org       = flag.String("user", "istio", "Github owner or org")
 	tokenFile = flag.String("token_file", "", "Github token file (optional)")
-	repo      = flag.String("repo", "istio", "Github repo")
+	repos     = flag.String("repos", "istio,api,proxy,test-infra", "Github repos, separate using \",\"")
```
Comment (rkpagadala) [inline toolbox/fixit/main.go]: One more repo 
istio.github.io

File: toolbox/fixit/main.go [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+	// Allowed field values can be found at https://developer.github.com/v3/search/#search-repositories
+	desc        = orderV("desc")
+	created     = sortV("created")
+	closedState = string("closed")
+)
+
+var (
+	org       = flag.String("user", "istio", "Github owner or org")
+	tokenFile = flag.String("token_file", "", "Github token file (optional)")
+	repos     = flag.String("repos", "istio,api,proxy,test-infra", "Github repos, separate using \",\"")
```
Comment (ldemailly) [inline toolbox/fixit/main.go]: can we add fortio too :-) ?

Comment (hklai) [reply toolbox/fixit/main.go]: Sure thing

File: toolbox/fixit/main.go (hunk 6 above)
Comment (hklai) [reply toolbox/fixit/main.go]: Added

File: toolbox/release_note_collector/main.go (hunk 4 above)
Comment (hklai) [reply toolbox/release_note_collector/main.go]: Done

File: toolbox/util/commonUtils.go (hunk 3 above)
Comment (hklai) [reply toolbox/util/commonUtils.go]: Done. Thanks!

File: toolbox/util/githubClient.go (hunk 5 above)
Comment (hklai) [reply toolbox/util/githubClient.go]: Done. Thanks!

File: toolbox/fixit/main.go (hunk 6 above)
Comment (ldemailly) [reply toolbox/fixit/main.go]: ty!

**Pull request merged**